views/building-hours/building-hours-helpers.js

```diff
@@ -0,0 +1,193 @@
+  }
+
+  let results = schedules.map(set => {
+    let prefix = set.title
+    let filteredSchedules = set.hours.filter(sched => sched.days.includes(dayOfWeek))
+    if (!filteredSchedules.length) {
+      return [[false, prefix, 'Closed today']]
+    }
+
+    return filteredSchedules.map(schedule => {
+      let isActive = isBuildingOpenAtMoment(schedule, m)
+      let status = formatBuildingTimes(schedule, m)
+      return [isActive, prefix, status]
+    })
+  })
+
+  return flatten(results)
+}
+
+export function getShortBuildingStatus(info: BuildingType, m: momentT): string {
+  let dayOfWeek = ((m.format('dd'): any): DayOfWeekEnumType)
+
+  let schedules = normalizeBuildingSchedule(info)
+  if (!schedules.length) {
+    return 'Closed'
+  }
+
+  let statuses = schedules.map(set => {
+    let filteredSchedules = set.hours.filter(sched => sched.days.includes(dayOfWeek))
+    if (!filteredSchedules.length) {
+      return 'Closed'
+    }
+
+    return filteredSchedules.map(schedule => getStatusOfBuildingAtMoment(schedule, m))
+  })
+
+  return flatten(statuses).find(status => status !== 'Closed') || 'Closed'
+}
+
+export function isBuildingOpen(info: BuildingType, m: momentT): boolean {
+  let dayOfWeek = ((m.format('dd'): any): DayOfWeekEnumType)
+
+  let schedules = normalizeBuildingSchedule(info)
+  if (!schedules.length) {
+    return false
+  }
+
+  let results = schedules.map(set => {
+    let filteredSchedules = set.hours.filter(sched => sched.days.includes(dayOfWeek))
+    if (!filteredSchedules.length) {
+      return false
+    }
+
+    return filteredSchedules.map(schedule => isBuildingOpenAtMoment(schedule, m))
+  })
+
+  return flatten(results).find(status => status !== false) || false
+}
+
+const daysOfTheWeek: DayOfWeekEnumType[] = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su']
+export function summarizeDays(days: DayOfWeekEnumType[]): string {
+  // If one day is given: return the full name of that day.
+  //    ['Fr'] => 'Friday'
+  // If multiple contiguous days are given: return the bookended 3-letter days
+  //    ['Mo', 'Tu', 'We'] => 'Mon — Wed'
+  // If multiple non-contiguous days are given: return the 2-letter days, comma-separated
+  //    ['Fr', 'Sa'] => 'Fr, Sa'
+  // If the span has a common shorthand: return that shorthand
+  //    ['Mo', 'Tu', 'We', 'Th', Fr'] => "Weekdays"
+
+  if (days.length === 1) {
+    return moment(days[0], 'dd').format('dddd')
+  }
+
+  // Sort the days so we have fewer edge-cases
+  let sortedDays = sortBy(days, d => daysOfTheWeek.indexOf(d))
+
+  let startDay = sortedDays[0]
+  let endDay = sortedDays[sortedDays.length-1]
+
+  // Get the indices of the start/end days from the master list of all days
+  let startIndex = daysOfTheWeek.indexOf(startDay)
+  let endIndex = daysOfTheWeek.indexOf(endDay)
+
+  // if the number of days given is not the number of days in the span,
+  // join the list. (There's no point to converting them here.)
+  if (endIndex - startIndex !== sortedDays.length - 1) {
```

Comment:
When I wrote that, I was half-thinking about other things, and it then took be a solid 20 minutes of thinking about it to understand it enough to write the comment… I kept missing the fact that the indices are taken from the master list of days, rather than the given set.